src/main/java/v/hryhoryk/onlinebookstore/controller/BookController.java

```diff
@@ -18,42 +23,92 @@
 import v.hryhoryk.onlinebookstore.dto.CreateBookRequestDto;
 import v.hryhoryk.onlinebookstore.service.BookService;
 
+@Tag(name = "Book store API",
+        description = "Endpoints for managing book store API")
 @RequiredArgsConstructor
 @RestController
 @RequestMapping(value = "/books")
 public class BookController {
     private final BookService bookService;
 
+    @Operation(summary = "Get all books",
+            description = "Returns paginated and sorted view of all books")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200",
+                    description = "Successfully retrieved"),
+            @ApiResponse(responseCode = "404",
+                    description = "Not found - no books are available")
+    })
     @ResponseStatus(HttpStatus.OK)
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.getAll();
+    public List<BookDto> getAll(Pageable pageable) {
+        return bookService.getAll(pageable);
     }
 
+    @Operation(summary = "Get a book by ID",
+            description = "Returns a book as per the ID")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200",
+                    description = "Successfully retrieved"),
+            @ApiResponse(responseCode = "404",
+                    description = "Not found - book by this ID was not found")
+    })
     @ResponseStatus(HttpStatus.OK)
     @GetMapping("/{id}")
     public BookDto getBookById(@PathVariable Long id) {
         return bookService.getById(id);
     }
 
+    @Operation(summary = "Search books by parameters",
+            description = "Returns books by certain searching parameters")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200",
+                    description = "Successfully retrieved"),
+            @ApiResponse(responseCode = "404",
+                    description = "Not found - books by this searching parameters were not found")
+    })
     @ResponseStatus(HttpStatus.OK)
     @GetMapping("/search")
-    public List<BookDto> search(BookSearchParameters searchParameters) {
-        return bookService.search(searchParameters);
+    public List<BookDto> search(BookSearchParameters searchParameters, Pageable pageable) {
```

Comment:
do you need to validate the search parameters?

Reply:
I guess no, we don't have any contsrataints validations in BookSearchParameters class

Reply:
I guess no, they can be dynamically changed, so they don't need any constraints